File: 7KYU/Stacked Balls - 2D/solution.c
#include <math.h>

double stack_height_2d(int n)
{
    if(n == 0)
      return 0.0;
    else
      return sqrt(pow((n-1.0), 2) - pow(((n-1.0)/2.0), 2))+1.0;      
}

File: 7KYU/Stacked Balls - 2D/solution.cs
using System;
public class Dinglemouse
{
    public static double StackHeight2d(int layers)
      =>  (layers == 0)
            ? 0d
            : Math.Sqrt(Math.Pow((layers-1.0), 2) - Math.Pow(((layers-1.0)/2.0), 2))+1.0;     
}